### ISSUE TITLE
make_zip.py had a bug when setting up full 64 bit distribution.

### ISSUE THREAD

**Issue by a28bdafa-f08b-4d5c-a058-5f14648d68ee** [python/cpython#74408]
BPO | [30222](https://bugs.python.org/issue30222)
--- | :---
Nosy | @pfmoore, @tjguk, @zware, @zooba, @AraHaan, @segevfiner
PRs | <li>python/cpython#2808</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = <Date 2017-07-27.18:21:21.214>
created_at = <Date 2017-05-01.18:59:18.240>
labels = ['3.7', 'invalid', 'OS-windows']
title = 'make_zip.py had a bug when setting up full 64 bit distribution.'
updated_at = <Date 2017-07-27.18:21:21.214>
user = 'https://github.com/AraHaan'
```

bugs.python.org fields:
```python
activity = <Date 2017-07-27.18:21:21.214>
actor = 'steve.dower'
assignee = 'none'
closed = True
closed_date = <Date 2017-07-27.18:21:21.214>
closer = 'steve.dower'
components = ['Demos and Tools', 'Windows']
creation = <Date 2017-05-01.18:59:18.240>
creator = 'Decorater'
dependencies = []
files = []
hgrepos = []
issue_num = 30222
keywords = []
message_count = 6.0
messages = ['292694', '298825', '298851', '298854', '298885', '299327']
nosy_count = 6.0
nosy_names = ['paul.moore', 'tim.golden', 'zach.ware', 'steve.dower', 'Decorater', 'Segev Finer']
pr_nums = ['2808']
priority = 'normal'
resolution = 'not a bug'
stage = 'resolved'
status = 'closed'
superseder = None
type = None
url = 'https://bugs.python.org/issue30222'
versions = ['Python 3.5', 'Python 3.6', 'Python 3.7']
```

</p></details>

**Comment by a28bdafa-f08b-4d5c-a058-5f14648d68ee** [comment]
Basically running make_zip like this:

".\\PCbuild\\amd64\\python.exe" ".\\Tools\\msi\\make_zip.py" -a x64 -o ".\\python36-x86-x64"

Cuases make_zip mess up where none of the libs\\*.lib, nor any of the assemblies are copied. And is reproducible on me.

However doing something like this on the 32 bit version of python does not get this issue.

**Comment by 6ccb8c66-b6ca-4c37-b449-e35b11a6b9e4** [comment]
Try with:

    .\python.bat Tools\msi\make_zip.py -a amd64 -o py37-amd64.zip

It requires amd64 instead of x64. Probably can add a choices (https://docs.python.org/3/library/argparse.html#choices) argument there to make it more robust.

**Comment by zooba** [comment]
I have a separate PR out that fixes this one, just waiting on Zach to review (but since it only touches the more obscure release files, it's rare for anyone but me to have an opinion). 

The script is supposed to take the build path rather than the architecture, but that got reverted at some point.

**Comment by 6ccb8c66-b6ca-4c37-b449-e35b11a6b9e4** [comment]
I did make a PR which only changes -a/--arch to a choices argument. https://github.com/python/cpython/pull/2808

I guess the PR you are talking about is https://github.com/python/cpython/pull/2750/files which changes this entirely.

**Comment by zooba** [comment]
That's the PR I meant (on phone today, so navigating gh is out). If you look through history you'll see it's mostly a revert of a revert, at least for that script.

**Comment by 6ccb8c66-b6ca-4c37-b449-e35b11a6b9e4** [comment]
Since [PR 2750](https://github.com/python/cpython/pull/2750) is merged, I think this can be closed.
